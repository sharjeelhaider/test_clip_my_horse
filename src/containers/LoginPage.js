import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {fetchData, receiveHorses} from '../actions/api';
import {horses} from '../utils/endpoints';
import {connect} from 'react-redux';
import {
  Image,
  StyleSheet,
  Text,
  TextInput,
  View,
  TouchableOpacity,
  ActivityIndicator,
  SafeAreaView,
} from 'react-native';
import {NavigationService} from '../navigation/NavigationService';

export class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
  }

  handleSubmit = (email, password) => {
    if (email && password) {
      this.props.getHorses(email, password);
    }
  };

  handleEmailText = e => {
    this.setState({
      email: e,
    });
  };

  handlePasswordText = e => {
    this.setState({
      password: e,
    });
  };

  render() {
    const {isLoading, isLoggedIn, error} = this.props.horses;
    if (isLoggedIn) {
      setTimeout(() => {
        NavigationService.navigate('Main');
      }, 100);
    }
    return (
      <View style={styles.container}>
        <SafeAreaView />
        <Image
          style={styles.logo}
          source={require('..//assets/ic_launcher.png')}
        />
        <Text style={styles.fieldTitle}>Email</Text>
        <TextInput
          style={styles.textInput}
          autoCapitalize="none"
          value={this.state.email}
          onChangeText={e => this.handleEmailText(e)}
        />
        <Text style={styles.fieldTitle}>Password</Text>
        <TextInput
          style={styles.textInput}
          secureTextEntry={true}
          autoCapitalize="none"
          value={this.state.password}
          onChangeText={e => this.handlePasswordText(e)}
        />
        <TouchableOpacity
          activeOpacity={0.8}
          style={styles.button}
          onPress={() =>
            this.handleSubmit(this.state.email, this.state.password)
          }>
          {isLoading ? (
            <ActivityIndicator color="#ffffff" size="small" />
          ) : (
            <Text style={styles.textButton}>Login</Text>
          )}
        </TouchableOpacity>
        {error ? <Text style={styles.fieldTitle}>{error.message}</Text> : null}
      </View>
    );
  }
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
    padding: 15,
  },
  textInput: {
    marginTop: 10,
    padding: 10,
    backgroundColor: '#fff',
    width: '100%',
  },
  logo: {
    width: 150,
    height: 150,
    alignSelf: 'center',
  },
  fieldTitle: {
    fontSize: 22,
    alignSelf: 'flex-start',
    marginTop: 10,
    color: 'white',
  },
  textButton: {
    fontSize: 22,
    alignSelf: 'center',
    color: 'white',
  },
  button: {
    marginTop: 20,
    backgroundColor: 'red',
    width: '100%',
    padding: 15,
  },
});

LoginPage.propTypes = {
  getHorses: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({horses: state.horses});
const mapDispatchToProps = dispatch => ({
  getHorses: (username, password) =>
    dispatch(fetchData(username, password, horses, receiveHorses)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoginPage);
