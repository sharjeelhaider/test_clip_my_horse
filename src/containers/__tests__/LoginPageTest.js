import React from 'react';
import {LoginPage} from '../LoginPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders a LoginPage using Snapshots', () => {
  const _getComponent = props => {
    return renderer.create(<LoginPage horses={props} getHorses={jest.fn()} />);
  };

  let component = _getComponent({isLoading: true});
  expect(component).toMatchSnapshot();

  component = _getComponent({isLoading: false});
  expect(component).toMatchSnapshot();
});

it('should change state if email entered', () => {
  const props = {horses: []};

  const wrapper = renderer.create(
    <LoginPage horses={props} getHorses={jest.fn()} />,
  );
  const instance = wrapper.getInstance();
  instance.handleEmailText('foo@gmail.com');
  expect(instance.state.email).toEqual('foo@gmail.com');
});

it('should change state if password entered', () => {
  const props = {horses: []};

  const wrapper = renderer.create(
    <LoginPage horses={props} getHorses={jest.fn()} />,
  );
  const instance = wrapper.getInstance();
  instance.handlePasswordText('bar');
  expect(instance.state.password).toEqual('bar');
});

it('should called getHorses function if credentials are passed to handleSubmit function', () => {
  const props = {horses: []};
  const wrapper = renderer.create(
    <LoginPage horses={props} getHorses={jest.fn()} />,
  );
  const instance = wrapper.getInstance();
  instance.handleSubmit('foo@gmail.com', 'bar');
  expect(instance.props.getHorses).toHaveBeenCalled();
});

it('should not called getHorses function if empty credentials are passed to handleSubmit function', () => {
  const props = {horses: []};
  const wrapper = renderer.create(
    <LoginPage horses={props} getHorses={jest.fn()} />,
  );
  const instance = wrapper.getInstance();
  instance.handleSubmit('', '');
  expect(instance.props.getHorses).not.toHaveBeenCalled();
});
