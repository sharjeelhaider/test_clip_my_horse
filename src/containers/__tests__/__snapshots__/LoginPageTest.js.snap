// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a LoginPage using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/ic_launcher.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 22,
        "marginTop": 10,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": 10,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 22,
        "marginTop": 10,
      }
    }
  >
    Password
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": 10,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "marginTop": 20,
        "opacity": 1,
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;

exports[`renders a LoginPage using Snapshots 2`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/ic_launcher.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 22,
        "marginTop": 10,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": 10,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 22,
        "marginTop": 10,
      }
    }
  >
    Password
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": 10,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "marginTop": 20,
        "opacity": 1,
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 22,
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
